lucene/test-framework/src/java/org/apache/lucene/analysis/ValidatingTokenFilter.java

```diff
@@ -50,6 +53,9 @@
   private final OffsetAttribute offsetAtt = getAttribute(OffsetAttribute.class);
   private final CharTermAttribute termAtt = getAttribute(CharTermAttribute.class);
 
+  // record all the Tokens seen so they can be dumped on failure
+  private final List<Token> tokens = new ArrayList<>();
```

Comment:
I slightly worry here that we are exposing ourselves to spurious out-of-memory failures during testRandomStrings() or similar; can we maybe limit this to just the last 200 tokens or something like that?

Reply:
This also needs to be emptied on `reset()`

Reply:
Oh I hadn't realized RandomChains could create such large strings that would be a concern!  Yes, I don't think it's helpful to have even 200 tokens printed out. In general your interest here will be in the final few tokens, so I think we can limit to 20 or so.